src/main.rs

```diff
@@ -0,0 +1,116 @@
+use std::env;
+use uuid::Uuid;
+use openssl::error::ErrorStack;
+use openssl::ssl::{SslConnector, SslMethod};
+use postgres::{error::SqlState, Client, Error, Transaction};
+use postgres_openssl::MakeTlsConnector;
+
+// BEGIN ssl_config
+fn ssl_config() -> Result<MakeTlsConnector, ErrorStack> {
+    let builder = SslConnector::builder(SslMethod::tls())?;
+    Ok(MakeTlsConnector::new(builder.build()))
+}
+// END ssl_config
+
+/// Runs op inside a transaction and retries it as needed.
+/// On non-retryable failures, the transaction is aborted and
+/// rolled back; on success, the transaction is committed.
+// BEGIN execute_txn
+fn execute_txn<T, F>(client: &mut Client, op: F) -> Result<T, Error>
+where
+    F: Fn(&mut Transaction) -> Result<T, Error>,
+{
+    let mut txn = client.transaction()?;
+    loop {
```

Comment:
It might be safer to repeat only a finite number of times.

Reply:
I didn't write this retry function, and am a little out of my depth on how to implement it differently. Maybe @ericharmeling can give some insight?

Reply:
Eric didn't write this code either. It was added roughly 5 years ago. I'm going to ask in #sql-experience.